lib/chef/chef_fs/file_system/repository/nodes_dir.rb

```diff
@@ -30,6 +31,24 @@ class NodesDir < Repository::Directory
           def make_child_entry(child_name)
             Node.new(child_name, self)
           end
+
+          def create_child(child_name, file_contents = nil)
+            child = super
+            File.chmod(0600, child.file_path)
+            if Chef::Platform.windows?
+              read_mask = Chef::ReservedNames::Win32::API::Security::GENERIC_READ
+              write_mask = Chef::ReservedNames::Win32::API::Security::GENERIC_WRITE
```

Comment:
I think you want FILE_GENERIC_READ and FILE_GENERIC_WRITE here, but I'm out the door. I can look after dinner.